### PR TITLE
Whitelist create environment endpoint 

### PR DESCRIPTION
**Pull request by medo-freiheit**

### CODE REVIEW

File: pkg/auth/azure_test.go [hunk 1]
```diff
@@ -247,35 +247,97 @@ func TestHttpMiddleware(t *testing.T) {
...
+			ExpectedError: "Failed to parse the JWT.\nError: token contains an invalid number of segments",
+			Authenticated: false,
+		},
+		{
+			Name:          "api call create environment - no env",
+			Path:          "/environments/",
+			Method:        http.MethodPost,
+			ExpectedError: "Failed to parse the JWT.\nError: token contains an invalid number of segments",
+			Authenticated: false,
 		},
```
Comment (jaypontes) [inline pkg/auth/azure_test.go:341]: Should there be a test with POST that is successful? Or is this not possible?

**Review by jaypontes — APPROVED** [review]

**Comment by leandro-freiheit** [comment]
I am not sure how the xpath function works, so I am not sure if I can properly review this.

File: pkg/auth/azure.go [hunk 2]
```diff
@@ -161,17 +161,26 @@ func HttpAuthMiddleWare(resp http.ResponseWriter, req *http.Request, jwks *keyfu
...
+	// Skip azure authentication with ID for `/` (POST: createEnv), `/release`, `/releasetrain` and `/locks`  endpoints. The requests will be validated with pgp signature
 	// usage in requests from outside the cluster (e.g. by GitHub Actions and the publish.sh script).
-	releaseTrainRx := regexp.MustCompile("^/environments/[^/]+/releasetrain$")
-	if releaseTrainRx.MatchString(req.URL.Path) {
-		return nil
-	}
-	environmentLocksRx := regexp.MustCompile("^/environments/[^/]+/locks/[^/]+$")
-	if environmentLocksRx.MatchString(req.URL.Path) {
-		return nil
+	group, tail := xpath.Shift(req.URL.Path)
```
Comment (leandro-freiheit) [inline pkg/auth/azure.go:167]: As I understand, this is shifting something in the request, but I don't understand what. Can you explain a bit @medo-freiheit?

Comment (jaypontes) [reply pkg/auth/azure.go:167]: From what I understand the shift simply splits the leftmost part of the URL
So if you have `/environments/prod/blabla`, the shift will `environments` part in the `group` variable and the `/prod/blabla` into the `tail` variable

Comment (leandro-freiheit) [reply pkg/auth/azure.go:167]: Alright, I am satisfied with the answer! Looks good to me

**Pull request merged**